File: Data Structures & Algorithms/counting-bits/submission-1.cpp
class Solution {

int fun(int n){
    int cnt = 0 ;
    while(n){
        if(1&n){
            cnt++ ;
        }
        n = (n>>1) ;
    }
    return cnt ;
}

public:
    vector<int> countBits(int n) {
        vector<int> ans(n+1,0) ;
        for(int i=1;i<=n;i++){
            ans[i] = fun(i) ;
        }
        return ans ;
    }
};

File: Data Structures & Algorithms/generate-parentheses/submission-1.py
class Solution:
    def generateParenthesis(self, n: int) -> List[str]:

        ans = set()

        def dfs(curr, open, closed):
            if open == 0 and closed == 0:
                ans.add(curr)
                return

            if open > 0:
                dfs(curr + "(", open-1, closed)
            
            
            if closed > open:
                dfs(curr + ")", open, closed-1)

            return
        
        dfs("", n, n)
        return list(ans)
        